bin/node/src/commands/block_producer.rs

```diff
@@ -105,6 +117,22 @@ impl BlockProducerCommand {
         // Start system monitor.
         SystemMonitor::new(monitor_interval).run_with_supervisor();
 
+        // Runtime validation for protocol constraints
+        if max_batches_per_block >= miden_objects::MAX_BATCHES_PER_BLOCK {
+            anyhow::bail!(
+                "max-batches-per-block ({}) cannot exceed protocol limit ({})",
+                max_batches_per_block,
+                miden_objects::MAX_BATCHES_PER_BLOCK
```

Comment:
Nit, the user should know what he configured already

```suggestion
                "max-batches-per-block cannot exceed protocol limit of {}",
                miden_objects::MAX_BATCHES_PER_BLOCK
```